src/utils/pars_utils.jl

```diff
@@ -41,45 +41,87 @@ function writevalidity(props_db::LegendDataManagement.MaybePropsDB, filekey::Fil
-        mkpath(dirname(validity_filename))
-        touch(validity_filename)
-        # check if validity already written
-        validity_lines = readlines(validity_filename)
-        # check if given validity already exists
-        is_validity = findall(x -> contains(x, "$pars_validTimeStamp") && contains(x, "$(string(category))"), validity_lines)
-        if isempty(is_validity)
-            if write_from_master @info "Write new validity for $pars_validTimeStamp" end
-            push!(validity_lines, "{\"valid_from\":\"$pars_validTimeStamp\", \"category\":\"$(string(category))\", \"apply\":[\"$(join(sort(apply), "\", \""))\"]}")
-        elseif length(is_validity) == 1
-            if write_from_master @info "Merge old $pars_validTimeStamp $(string(category)) validity entry" end
-            apply = unique(append!(Vector{String}(JSON.parse(validity_lines[first(is_validity)])["apply"]), apply))
-            validity_lines[first(is_validity)] = "{\"valid_from\":\"$pars_validTimeStamp\", \"category\":\"$(string(category))\", \"apply\":[\"$(join(sort(apply), "\", \""))\"]}"
+        # locate & create file if missing
+        dst = joinpath(data_path(props_db), "validity.yaml")
+        mkpath(dirname(dst))
+        if !isfile(dst)
+            open(dst, "w") do io
+                println(io, "[]")
+            end
         end
-        # write validity
-        open(validity_filename, "w") do io
-            for line in sort(validity_lines)
-                println(io, line)
+
+        # load or initialize the vector of entries
+        raw     = YAML.load_file(dst)
+        entries = raw isa Vector ? raw : Any[]
+
+        ts = string(filekey.time)
+
+        # drop any previous entry with the same timestamp
+        filter!(e -> e["valid_from"] != ts, entries)
+
+        # build a Dict containing the validity to be saved
+        entry = Dict{String,Any}()
+        entry["valid_from"] = ts
+        entry["apply"]      = apply
+        entry["category"]   = category
+        entry["mode"]       = mode
+        push!(entries, entry)
+        
+        write_from_master && @info "Write validity for $ts"
+
+        # sort the entries by timestamp
+        sort!(entries, by = e -> e["valid_from"])
+
+        function write_entries_ordered(dst, entries)
+            open(dst, "w") do io
+                for e in entries
+                    println(io, "- valid_from: ", e["valid_from"])
+
+                    println(io, "  apply:")
+                    for path in e["apply"]
+                        println(io, "    - ", path)
+                    end
+
+                    println(io, "  category: ", e["category"])
+                    println(io, "  mode: ", e["mode"])
+                    println(io)  # newline between entries
+                end
             end
         end
+
+        # emit valid YAML, with each Dict’s keys in insertion order
+        write_entries_ordered(dst, entries)
     end
 end
```

Comment:
Why define a function here and call it, when you could just call the code ?

Also, is there no elegant way using `YAML.jl` to write these entries instead of using `println(io, ...)`?
Not that I have tested this, but I would expect something along the lines of


```suggestion

        # load or initialize the vector of entries
        raw     = YAML.load_file(dst, dicttype = OrderedDict)
        entries = raw isa Vector ? raw : Any[]

        ts = string(filekey.time)

        # drop any previous entry with the same timestamp
        filter!(e -> e["valid_from"] != ts, entries)

        # build a Dict containing the validity to be saved
        push!(entries, Dict{String,Any}(
            "valid_from" => ts,
            "apply" => apply,
            "category" => category,
            "mode" => mode
        ))
        
        write_from_master && @info "Write validity for $ts"

        # sort the entries by timestamp
        sort!(entries, by = e -> e["valid_from"])
        
        # create a new file with the same filename with updated entries
        YAML.write_file(dst, entries)
end
```